# MGnify_BProjekt - [MGnify Pipeline](https://www.ebi.ac.uk/metagenomics/pipelines/5.0#raw)

For my Bachelor's project, I am undertaking the task of porting, enhancing, and benchmarking a read-based MGnify analysis pipeline designed for microbiome data analysis, utilizing the Galaxy framework.

## Main Tasks:

### Mgnify Pipeline Workflow

- [x]    Get familiar with the Mgnify pipeline
- [x]   Run the Mgnify pipeline
- [x]  Identify the steps and tools in the pipeline [Here](https://docs.google.com/spreadsheets/d/1bZsgFdGKkLrX-ie84RdUsyuaVI_wCFtJdCVprq7y6dk/edit#gid=1401173895)
- [ ] Integrate the missing tools in Galaxy using Planemo (https://planemo.readthedocs.io/en/latest/)
- [ ] Build in Galaxy the pipeline connecting the tools
- [ ]  Benchmark the Galaxy pipeline against the original one
- [ ]  Annotate and submit the pipeline to Galaxy's Intergalactic Workflow Commission (https://github.com/galaxyproject/iwc/)

## Overview

### Complete the MGnify Amplicon Pipeline 5.0 in the Galaxy Framework

#### Tasks:
- [ ] Port the existing MGnify Amplicon Pipeline 5.0 to the Galaxy framework.


### Work on the MGnify Raw reads Pipeline 5.0.

#### Tasks:

##### Dependencies Analysis
- [x] List the identified dependencies in a separate sheet for reference.

##### Galaxy Availability Analysis
- [x] Check whether each tool used in the MGnify Amplicon Pipeline 5.0 is available within the Galaxy framework.
- [x] Mark the availability status for each tool (Available / Not Available).

##### Bioconda Availability Analysis
- [x] Verify the availability of the tools used in the MGnify Amplicon Pipeline 5.0 in the Bioconda repository.
- [x] Mark the availability status for each tool in Bioconda (Available / Not Available).

##### Create Documentation
- [x] Document the findings of the dependencies and tool availability analyses in a Sheets document for reference.

### Work on implementing the mOTUs3 tool - https://github.com/motu-tool/mOTUs

1. [ ] Clone the mOTUs3 repository from GitHub to your development environment.
2. [ ] Review the mOTUs3 documentation to understand its installation and usage requirements.
3. [ ] Install any necessary dependencies and prerequisites for mOTUs3 as per the documentation.
4. [ ] Build or compile the mOTUs3 tool according to the provided instructions.
5. [ ] Test the mOTUs3 tool in your local environment to ensure it functions correctly.

Here are tasks for "Wrapper - Tutorial," "Datenmanager - Tutorial," and "Benchmark Data":

### Wrapper - Tutorial

1. [ ] Develop a comprehensive tutorial for the Wrapper component of the MGnify Amplicon Pipeline 5.0. Include step-by-step instructions for users.
2. [ ] Create example datasets or inputs that users can use to practice with the Wrapper.
3. [ ] Provide explanations and context for each step in the tutorial.
4. [ ] Ensure the tutorial is well-documented and user-friendly.

### Datenmanager - Tutorial

1. [ ] Develop a tutorial for the Datenmanager component of the MGnify Amplicon Pipeline 5.0, detailing its functionalities and usage.
2. [ ] Include practical examples of how to manage data effectively with the Datenmanager.
3. [ ] Clarify any potential issues or common mistakes users might encounter and provide solutions.
4. [ ] Ensure the tutorial is clear and easy to follow.
